workspaces/adr/plugins/adr-backend/src/service/router.test.ts

```diff
@@ -228,4 +232,72 @@ describe('createRouter', () => {
       expect(fileTwoBody.data).toBe(testFileTwoContent);
     });
   });
+
+  describe(`GET ${imageEndpointName}`, () => {
+    it('returns bad request (400) when no url is provided', async () => {
+      const urlNotSpecifiedRequest = await request(app).get(imageEndpointName);
+      const urlNotSpecifiedStatus = urlNotSpecifiedRequest.status;
+      const urlNotSpecifiedMessage = urlNotSpecifiedRequest.body.message;
+
+      const urlNotFilledRequest = await request(app).get(
+        `${imageEndpointName}?url=`,
+      );
+      const urlNotFilledStatus = urlNotFilledRequest.status;
+      const urlNotFilledMessage = urlNotFilledRequest.body.message;
+
+      const expectedStatusCode = 400;
+      const expectedErrorMessage = 'No URL provided';
+
+      expect(urlNotSpecifiedStatus).toBe(expectedStatusCode);
+      expect(urlNotSpecifiedMessage).toBe(expectedErrorMessage);
+
+      expect(urlNotFilledStatus).toBe(expectedStatusCode);
+      expect(urlNotFilledMessage).toBe(expectedErrorMessage);
+    });
+
+    it('returns bad request (400) when unsupported image format is provided', async () => {
+      const urlNotFilledRequest = await request(app).get(
+        `${imageEndpointName}?url=testImage.txt`,
+      );
+      const urlNotFilledStatus = urlNotFilledRequest.status;
+      const urlNotFilledMessage = urlNotFilledRequest.body.message;
+
+      const expectedStatusCode = 400;
+      const expectedErrorMessage = 'Image type txt is not supported';
+
+      expect(urlNotFilledStatus).toBe(expectedStatusCode);
+      expect(urlNotFilledMessage).toBe(expectedErrorMessage);
+    });
+
+    it('returns the correct image when reading a url', async () => {
```

Comment:
Mind also adding a test for the cached scenario?

Reply:
I'm having a hard time adding a test for cached scenario because I don't have an `etag`.
I'm not too fluent with `request(app).get()` from `supertest` and I don't know how can I mock it so I can get to the cached scenario.
https://github.com/backstage/community-plugins/blob/b6aaaa48afb1b6733b1346b11bd42f2f42efd13f/workspaces/adr/plugins/adr-backend/src/service/router.test.ts#L281-L283

Would you be able to help me on this one?

Reply:
We can leverage our mocked cache client and url reader in our test environment to simulate the etag behavior - you should be able to spyOn the the `readUrl` method and explicitly throw a `NotModifiedError` and similarly you can also spyOn the .`get` method of the cacheClient to return mocked cached data. Your test would then confirm the response contains the expected data which would test the cached scenario behaves as expected

Reply:
Thank you for your explanation. :slightly_smiling_face: 
I've added test scenario for cached image.